digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/process/SpeedLimitValidator.scala

```diff
@@ -0,0 +1,114 @@
+
+                case TrafficDirection.AgainstDigitizing =>
+                  trafficSignService.getTrafficSignByRadius(last, 50, Some(SpeedLimits))
+                    .filter (trafficSign => SideCode.apply(trafficSign.validityDirection) == TrafficDirection.toSideCode(TrafficDirection.AgainstDigitizing))
+
+                case _ =>
+              trafficSignService.getTrafficSignByRadius(first, 50, Some(SpeedLimits)).filter(_.validityDirection == SideCode.TowardsDigitizing.value) ++
+                trafficSignService.getTrafficSignByRadius(last, 50, Some(SpeedLimits)).filter(_.validityDirection == SideCode.AgainstDigitizing.value)
+
+          }}).filter(_.linkId == speedLimit.linkId)
+
+    if (trafficSingsByRadius.nonEmpty) {
+      (trafficSingsByRadius.groupBy(minDistance(_, first)) ++ trafficSingsByRadius.groupBy(minDistance(_, last))).reduceLeft(min)._2
+    }else
+      Seq()
+  }
+
+  private def speedLimitValueValidator(speedLimit: SpeedLimit, trafficSign: PersistedTrafficSign): Option[SpeedLimit] = {
+    val startUrbanAreaSpeedLimit: Int = 50
+    val endUrbanAreaSpeedLimit: Int = 80
+
+    speedLimit.value match {
+      case Some(NumericValue(speedLimitValue)) =>
+        TrafficSignType.apply(trafficSign.propertyData.find(p => p.publicId == "trafficSigns_type").get.values.head.propertyValue.toInt) match {
+          case TrafficSignType.SpeedLimit | TrafficSignType.SpeedLimitZone =>
+            trafficSign.propertyData.find(p => p.publicId == "trafficSigns_value").get.values.headOption match {
+              case Some(trafficSignValue) if trafficSignValue.propertyValue != speedLimitValue.toString =>
+                Some(speedLimit)
+              case _ => None
+            }
+
+          case TrafficSignType.EndSpeedLimitZone | TrafficSignType.EndSpeedLimit =>
+            trafficSign.propertyData.find(p => p.publicId == "trafficSigns_value").get.values.headOption match {
+              case Some(trafficSignValue) if trafficSignValue.propertyValue == speedLimitValue.toString =>
+                Some(speedLimit)
+              case _ => None
+            }
+
+          case TrafficSignType.UrbanArea if speedLimitValue != startUrbanAreaSpeedLimit =>
+            Some(speedLimit)
+
+          case TrafficSignType.EndUrbanArea if speedLimitValue != endUrbanAreaSpeedLimit =>
+            Some(speedLimit)
+
+          case _ => None
+        }
+      case _ => None
+    }
+  }
+
+  private def filterBySide(trafficSign: PersistedTrafficSign, speedLimit: SpeedLimit, roadLink: RoadLink) : Boolean = {
+    speedLimit.sideCode match {
+      case SideCode.TowardsDigitizing | SideCode.AgainstDigitizing =>
+        trafficSign.validityDirection == speedLimit.sideCode.value
+      case _ =>
+        if (roadLink.trafficDirection == TrafficDirection.BothDirections)
+          true
+        else
+          TrafficDirection.toSideCode(roadLink.trafficDirection) == SideCode.apply(trafficSign.validityDirection)
```

Comment:
This is the same as return `roadLink.trafficDirection == TrafficDirection.BothDirections || TrafficDirection.toSideCode(roadLink.trafficDirection) == SideCode.apply(trafficSign.validityDirection)`